File: ArchLucid.Application/Billing/TenantCostEstimate.cs
using ArchLucid.Core.Tenancy;

namespace ArchLucid.Application.Billing;

/// <summary>Rough monthly spend band for operator settings (Standard+ only at the HTTP layer).</summary>
public sealed record TenantCostEstimate(
    string Currency,
    TenantTier Tier,
    decimal EstimatedMonthlyUsdLow,
    decimal EstimatedMonthlyUsdHigh,
    IReadOnlyList<string> Factors,
    string MethodologyNote);

File: archlucid-ui/src/types/tenant-cost-estimate.ts
import type { components } from "@/lib/openapi-schemas";

type TenantCostEstimateResponseSchema = components["schemas"]["TenantCostEstimateResponse"];

export type TenantCostEstimateResponse = TenantCostEstimateResponseSchema &
  Required<
    Pick<
      TenantCostEstimateResponseSchema,
      | "currency"
      | "estimatedMonthlyUsdLow"
      | "estimatedMonthlyUsdHigh"
      | "factors"
      | "methodologyNote"
      | "tier"
    >
  >;
